apps/prairielearn/src/components/QuestionScore.html.ts

```diff
@@ -0,0 +1,470 @@
+import { escapeHtml, html } from '@prairielearn/html';
+
+import type {
+  Assessment,
+  AssessmentInstance,
+  AssessmentQuestion,
+  InstanceQuestion,
+  Question,
+  Submission,
+  Variant,
+} from '../lib/db-types.js';
+import { formatPoints, formatPointsOrList } from '../lib/format.js';
+import { idsEqual } from '../lib/id.js';
+
+export function QuestionScorePanel({
+  instance_question,
+  assessment,
+  assessment_question,
+  question,
+  assessment_instance,
+  instance_question_info,
+  variant,
+  no_variant_exists,
+  authz_result,
+  csrfToken,
+  urlPrefix,
+}: {
+  instance_question: InstanceQuestion & {
+    allow_grade_left_ms?: number;
+    allow_grade_interval?: string;
+  };
+  assessment: Assessment;
+  assessment_question: AssessmentQuestion;
+  question: Question;
+  assessment_instance: AssessmentInstance;
+  variant: Variant;
+  instance_question_info: {
+    question_number: string;
+    previous_variants:
+      | (Variant & { max_submission_score: NonNullable<Submission['score']> })[]
+      | null;
+  };
+  no_variant_exists?: boolean;
+  authz_result?: { authorized_edit: boolean | null };
+  csrfToken: string;
+  urlPrefix: string;
+}) {
+  const hasAutoAndManualPoints =
+    assessment_question.max_auto_points &&
+    (assessment_question.max_manual_points ||
+      instance_question.manual_points ||
+      instance_question.requires_manual_grading);
+
+  return html`
+    <div class="card mb-4" id="question-score-panel">
+      <div class="card-header bg-secondary text-white">
+        Question ${instance_question_info.question_number}
```

Comment:
The original EJS had a bug here that seems to have been going on for a long time. This was using `instance_question.question_number`, which is undefined. As a result, this component did not have the question number listed in the code. I have added this back, assuming this is indeed the expected functionality.